dotnet/samples/Demos/StepwisePlannerMigration/Controllers/AutoFunctionCallingController.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Connectors.OpenAI;
+using StepwisePlannerMigration.Models;
+using StepwisePlannerMigration.Plugins;
+using StepwisePlannerMigration.Services;
+
+namespace StepwisePlannerMigration.Controllers;
+
+/// <summary>
+/// This controller shows a new recommended approach how to use planning capability by using Auto Function Calling.
+/// </summary>
+[ApiController]
+[Route("auto-function-calling")]
+public class AutoFunctionCallingController : ControllerBase
+{
+    private readonly Kernel _kernel;
+    private readonly IChatCompletionService _chatCompletionService;
+    private readonly IPlanProvider _planProvider;
+
+    public AutoFunctionCallingController(
+        Kernel kernel,
+        IChatCompletionService chatCompletionService,
+        IPlanProvider planProvider)
+    {
+        this._kernel = kernel;
+        this._chatCompletionService = chatCompletionService;
+        this._planProvider = planProvider;
+
+        this._kernel.ImportPluginFromType<TimePlugin>();
+        this._kernel.ImportPluginFromType<WeatherPlugin>();
+    }
+
+    /// <summary>
+    /// Action to generate a plan. Generated plan will be populated in <see cref="ChatHistory"/> object.
+    /// </summary>
+    [HttpPost, Route("generate-plan")]
+    public async Task<IActionResult> GeneratePlanAsync(PlanRequest request)
+    {
+        ChatHistory chatHistory = [];
+        chatHistory.AddUserMessage(request.Goal);
+
+        OpenAIPromptExecutionSettings executionSettings = new() { ToolCallBehavior = ToolCallBehavior.AutoInvokeKernelFunctions };
```

Comment:
When generating a plan I'd expect No functions to be invoked i.e. using the `none` option.

Reply:
I tried to use this approach - the model returns just first function and waits for its execution and result. So, users will have to do that manually. But with `ToolCallBehavior.AutoInvokeKernelFunctions` plan will be generated and executed step by step, and it's possible to get a result, and generated plan in ChatHistory to cache and re-execute it later if needed.